frontend/app/components/transportation/study-area-map/census-tract-selector.js

```diff
@@ -0,0 +1,49 @@
+import Component from '@ember/component';
+
+/**
+ * This component, given a transportation analysis model (analysis), and selected feature array
+ * (from Mapbox::FeatureSelector) will handle adding or removing the given feature, identified by 
+ * the 'geoid' property, to the analysis' jtwStudySelection property
+ */
+export default class TransportationStudyAreaMapCensusTractSelectorComponent extends Component {
+  /**
+   * The transportation-analysis Model, passed down from projects/show/transportation-analysis controller
+   */
+  analysis = {};
+
+  /**
+   * Array containing selected feature passed down from the Mapbox::FeatureSelector component.
+   * Can be expected to only contain a single Feature object, as census tracts do not overlap,
+   * so multiple census tracts cannot intersect a single click event point
+   */
+  selectedFeatureArray = [];
+
+
+  /**
+   * Census tract selection handling is triggered whenever bound property 'selectedFeatures' is updated
+   */
+  didUpdateAttrs() {
+    if(this.selectedFeatureArray) {
```

Comment:
Because the default value of `selectedFeatureArray` is `[]`, this will always be true (unless it is set to null later). Why do we need this defensive check here?